providers/common/ciphers/aes.c

```diff
@@ -443,35 +444,36 @@ static int aes_ctx_set_params(void *vctx, const OSSL_PARAM params[])
         { OSSL_FUNC_CIPHER_CTX_GET_PARAMS, (void (*)(void))aes_ctx_get_params }, \
         { OSSL_FUNC_CIPHER_CTX_SET_PARAMS, (void (*)(void))aes_ctx_set_params }, \
         { 0, NULL } \
-    };
+    }; \
+    const char *aes##keylen##mode##_names[]
 
 /* ECB */
-IMPLEMENT_block_funcs(ecb, 256, 0)
-IMPLEMENT_block_funcs(ecb, 192, 0)
-IMPLEMENT_block_funcs(ecb, 128, 0)
+IMPLEMENT_block_funcs(ecb, 256, 0) = { "AES-256-ECB", "aes-256-ecb", NULL };
+IMPLEMENT_block_funcs(ecb, 192, 0) = { "AES-192-ECB", "aes-192-ecb", NULL };
+IMPLEMENT_block_funcs(ecb, 128, 0) = { "AES-128-ECB", "aes-128-ecb", NULL };
 
 /* CBC */
-IMPLEMENT_block_funcs(cbc, 256, 16)
-IMPLEMENT_block_funcs(cbc, 192, 16)
-IMPLEMENT_block_funcs(cbc, 128, 16)
+IMPLEMENT_block_funcs(cbc, 256, 0) = { "AES-256-CBC", "aes-256-cbc", NULL };
+IMPLEMENT_block_funcs(cbc, 192, 0) = { "AES-192-CBC", "aes-192-cbc", NULL };
```

Comment:
The 16 values have been replaced with 0 here but not below?
It doesn't look like the value is used in the macros but consistency seems worthwhile.

Reply:
The values *is* used...

````
        { OSSL_FUNC_CIPHER_IV_LENGTH, (void (*)(void))iv_length_##ivlen }, \
````

Reply:
Oops, the 0s are wrong, of course...  Will fix